timemachine/cpp/src/potential.cu

```diff
@@ -23,66 +25,63 @@ void Potential::execute_host(
 
     const int &D = Potential::D;
 
-    double *d_x;
-    double *d_p;
-    double *d_box;
-
-    gpuErrchk(cudaMalloc(&d_x, N * D * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_x, h_x, N * D * sizeof(double), cudaMemcpyHostToDevice));
-
-    gpuErrchk(cudaMalloc(&d_p, P * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_p, h_p, P * sizeof(double), cudaMemcpyHostToDevice));
+    DeviceBuffer<double> d_x(N * D);
+    DeviceBuffer<double> d_p(P);
+    DeviceBuffer<double> d_box(D * D);
 
-    gpuErrchk(cudaMalloc(&d_box, D * D * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_box, h_box, D * D * sizeof(double), cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_x.data, h_x, d_x.size, cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_p.data, h_p, d_p.size, cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_box.data, h_box, d_box.size, cudaMemcpyHostToDevice));
```

Comment:
Would it be useful to move this line into a `DeviceBuffer::to_device` method or similar?

Reply:
Nice idea! I implemented `DeviceBuffer::copy_from` and `DeviceBuffer::copy_to` in 8d982bf, but am still undecided as to whether I actually like these names better than `to_device` and `to_host` (I guess I found the latter slightly awkward as members of `DeviceBuffer`, but the explicitness is also nice)

Reply:
I prefer the way you did it: `copy_from` / `copy_to`!